ratel/src/lexer/mod.rs

```diff
@@ -417,6 +417,7 @@ const SLH: ByteHandler = Some(|lex| {
                 match lex.next_byte() {
                     b'*' => {
                         match lex.next_byte() {
+                            b'*' => lex.retreat(),
```

Comment:
I assume there was an issue with `**/` endings? In general I like to keep both the lexer and the parser predictive, that is to only ever read a single byte once and only consume a token once. There should be a way of doing this without introducing backtracking to the lexer.

I'll try to suggest a solution later after I've pondered on it a bit :).

Reply:
Ok, found a non-backtracking solution:

```rust
        // block comment
        b'*' => {
            lex.bump(); // added this

            // Keep consuming bytes until */ happens in a row
            unwind_loop!({
                match lex.read_byte() { // `read_byte` instead of `next_byte`, so `index` isn't incremented
                    b'*' => {
                        match lex.next_byte() {
                            b'/' => {
                                lex.bump();
                                return lex.consume();
                            },
                            0 => return lex.token = UnexpectedEndOfProgram,
                            _ => {}
                        }
                    },
                    0 => return lex.token = UnexpectedEndOfProgram,
                    _ => lex.bump() // increment the index here
                }
            });
        },
```

Reply:
@maciejhirsz Your solution can not parse:

1. `/**/`
2. `/* ***/`
3. `/* **/`
4. `/****/`
5. `/**  **/`
6. ...

```rust
        // block comment
        b'*' => {
            lex.bump();
            unwind_loop!({
                match lex.next_byte() {
                    b'*' => {
                       /// WARN: The point is `next_byte()` function has already bump,
                        match lex.next_byte() {
                            b'/' => {
                                lex.bump();
                                return lex.consume();
                            },
                            0 => return lex.token = UnexpectedEndOfProgram,
                            _ => {}
                        }
                    },
                    0 => return lex.token = UnexpectedEndOfProgram,
                    _ => lex.bump()
                }
            });
        },


```

Reply:
The first `next_byte` in a loop needs to change to `read_byte`. I've changed 3 lines in the code above, all have comments to why :).

Reply:
@maciejhirsz it work :)